File: app/src/main/java/org/vaccinationreminder/MainFragmentHolder.java
package org.vaccinationreminder;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.NavigationView;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import java.util.Objects;

public class MainFragmentHolder extends AppCompatActivity {

    DrawerLayout HomeActivityNavDrawer;

    @Override

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fragment_holder);


        Toolbar toolbar = findViewById(R.id.activity_home_toolbar);
        setSupportActionBar(toolbar);

        ActionBar actionbar = getSupportActionBar();
        Objects.requireNonNull(actionbar).setDisplayHomeAsUpEnabled(true);
        Objects.requireNonNull(actionbar).setHomeAsUpIndicator(R.drawable.ic_menu);

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        final HomeFragment homeFragment = new HomeFragment();
        final AddNewChildFragment childManagerFragment = new AddNewChildFragment();

        fragmentTransaction.add(R.id.fragment_holder, homeFragment, "home");
        fragmentTransaction.commit();

        HomeActivityNavDrawer = findViewById(R.id.activity_home_drawer_layout);
        NavigationView HomeActivityNavView = findViewById(R.id.activity_home_nav_menu);
        HomeActivityNavView.setCheckedItem(R.id.home_fragment);

        HomeActivityNavView.setNavigationItemSelectedListener(
                new NavigationView.OnNavigationItemSelectedListener() {
                    @Override
                    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {

                        FragmentManager fragmentManager = getSupportFragmentManager();
                        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

                        menuItem.setChecked(true);
                        HomeActivityNavDrawer.closeDrawers();

                        if (menuItem.getItemId() == R.id.home_fragment) {
                            fragmentTransaction.replace(R.id.fragment_holder, homeFragment, "home");
                        }

                        if (menuItem.getItemId() == R.id.child_manager) {
                            fragmentTransaction.replace(R.id.fragment_holder, childManagerFragment, "childManager");
                        }
                        fragmentTransaction.commit();
                        return true;
                    }
                });

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                HomeActivityNavDrawer.openDrawer(GravityCompat.START);
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/adityasubathu/vaccinationreminder/MainFragmentHolder.kt
package com.adityasubathu.vaccinationreminder

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView

class MainFragmentHolder : AppCompatActivity() {

    private lateinit var homeActivityNavDrawer: DrawerLayout

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_fragment_holder)

        val toolbar = findViewById<Toolbar>(R.id.activity_home_toolbar)
        setSupportActionBar(toolbar)

        val actionbar = supportActionBar
        actionbar!!.setDisplayHomeAsUpEnabled(true)
        actionbar.setHomeAsUpIndicator(R.drawable.ic_menu)

        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        val homeFragment = HomeFragment()
        val childManagerFragment = AddNewChildFragment()
        val vaccineInfoFragment = VaccineInfoFragment()
        val alarmManagerFragment = AlarmManagerFragment()

        fragmentTransaction.add(R.id.fragment_holder, homeFragment, "home")
        fragmentTransaction.commit()

        homeActivityNavDrawer = findViewById(R.id.activity_home_drawer_layout)
        val homeActivityNavView = findViewById<NavigationView>(R.id.activity_home_nav_menu)
        homeActivityNavView.setCheckedItem(R.id.home_fragment)
        homeActivityNavView.setNavigationItemSelectedListener { menuItem ->
            val fm = supportFragmentManager
            val ft = fm.beginTransaction()
            menuItem.isChecked = true
            homeActivityNavDrawer.closeDrawers()
            if (menuItem.itemId == R.id.home_fragment) {
                ft.replace(R.id.fragment_holder, homeFragment, "home")
            }
            if (menuItem.itemId == R.id.child_manager) {
                ft.replace(R.id.fragment_holder, childManagerFragment, "childManager")
            }
            if (menuItem.itemId == R.id.vaccines_info) {
                ft.replace(R.id.fragment_holder, vaccineInfoFragment, "vaccineInfo")
            }
            if (menuItem.itemId == R.id.alarm_manager) {
                ft.replace(R.id.fragment_holder, alarmManagerFragment, "alarmManager")
            }
            ft.commit()
            true
        }
    }

    fun setActionBarTitle(title: String?) {
        supportActionBar!!.title = title
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            homeActivityNavDrawer.openDrawer(GravityCompat.START)
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}